core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
Sorry, @jenswi-linaro I misunderstood your point. :)
Yes, we can remove THREAD_EXCP_IRQ/FIQ/ABT.
As we still need the bit masks for the exceptions. So, my suggestion is that we can just keep these macros:
THREAD_EXCP_FOREIGN_INTR
THREAD_EXCP_NATIVE_INTR
THREAD_EXCP_ALL

And define the bit masks for FIQ/IRQ/ABT with the same index (need shift when use) of ARM archtecture. e.g.
#define THREAD_MASK_FIQ (1 << 0)
#define THREAD_MASK_IRQ (1 << 1)
#define THREAD_MASK_ABT (1 << 1)
These macros are only used by THREAD_EXCP_*. They are more explicit than just using the number. (We can work out a better prefix if you are all happy with this idea)
#ifdef CFG_GICV3
#define THREAD_EXCP_FOREIGN_INTR THREAD_MASK_FIQ
#else
#define THREAD_EXCP_FOREIGN_INTR THREAD_MASK_IRQ
#endif
...